RecoLocalTracker/SiStripRecHitConverter/interface/StripCPE.h

```diff
@@ -28,6 +29,45 @@ class StripCPE : public StripClusterParameterEstimator
 	    const SiStripLatency&);    
   LocalVector driftDirection(const StripGeomDetUnit* det) const;
 
+ struct Param {
+    Param() : topology(nullptr) {}
+    StripTopology const * topology;
+    LocalVector drift;
+    float thickness, invThickness,pitch_rel_err2, maxLength;
+    int nstrips;
+    float backplanecorrection;
+    SiStripDetId::ModuleGeometry moduleGeom;
+    float coveredStrips(const LocalVector&, const LocalPoint&) const;
+  };
+
+  
+  struct AlgoParam {
+    Param const & p; const LocalTrajectoryParameters & ltp;
+    SiStripDetId::SubDetector loc; float afullProjection; float corr;
+  };
+
+
+  virtual StripClusterParameterEstimator::LocalValues
+  localParameters( const SiStripCluster& cl, AlgoParam const & ap) const {
+    return std::make_pair(LocalPoint(), LocalError());
+  }
+  
+  AlgoParam getAlgoParam(const GeomDetUnit& det, const LocalTrajectoryParameters & ltp) const {
+
+    StripCPE::Param const & p = param(det);
+    SiStripDetId::SubDetector loc = SiStripDetId( det.geographicalId() ).subDetector();  
+ 
+    LocalVector track = ltp.momentum();
+    track *= -p.thickness/track.z();
```

Comment:
@VinInn: I was thinking of #14306, which is not related to the code in this PR, but fixes a bug with a bad TrajectoryStateOnSurface. I have a faint recollection of seeing similar issues previously.

How many paths through the code lead to line 61 above? If you have an idea of how to find them all, we could check that the proper validation is being done in advance. But I would prefer bullet-proof code that never has a possibility to propagate NANs and nonsense values.